core/src/jvmMain/kotlin/com/ustadmobile/core/util/ShrinkUtils.kt

```diff
@@ -0,0 +1,98 @@
+package com.ustadmobile.core.util
+
+import com.github.aakira.napier.Napier
+import com.ustadmobile.core.catalog.contenttype.VideoTypePlugin
+import java.io.BufferedReader
+import java.io.File
+import java.io.IOException
+import java.io.InputStreamReader
+
+object ShrinkUtils {
+
+    fun findInPath(commandName: String, systemPropName: String? = null): String? {
+
+        if(systemPropName != null){
+            return if(File(systemPropName).exists()){
+                systemPropName
+            }else{
+                null
+            }
+        }
+
+        return System.getenv("PATH").split(File.pathSeparator).firstOrNull() {
+            File(it, commandName).exists()
+        }?.let { File(it, commandName).path }
+    }
+
+    fun getVideoResolutionMetadata(srcFile: File): Triple<Int, Int, String> {
+
+        val ffprobePath = findInPath("ffprobe")
```

Comment:
This needs to throw an exception if ffprobe of ffmpeg cannot be found. 

As we are using these for all media conversion I think we should add a warning in UmRestApplication if ffprobe and ffmpeg are not available (so people should see it as soon as they start the server, rather than getting very confused about why media conversion isn't working).